sandbox/src/components/InAppMessagesDemo/InAppMessages.js

```diff
@@ -66,59 +51,27 @@ if (alloyInstance !== window.alloy) {
   });
 }
 
-alloyInstance("subscribeRulesetItems", {
-  surfaces: [surface],
-  callback: result => {
-    console.log("subscribeRulesetItems", result);
-  }
-});
-
-const uuidv4 = () => {
-  return ([1e7] + -1e3 + -4e3 + -8e3 + -1e11).replace(/[018]/g, c =>
-    (
-      c ^
-      (crypto.getRandomValues(new Uint8Array(1))[0] & (15 >> (c / 4)))
-    ).toString(16)
-  );
-};
-
-const getInAppPayload = async payload => {
-  const res = await fetch(
-    `https://${edgeDomain}/ee/or2/v1/interact?configId=${datastreamId}&requestId=${uuidv4()}`,
-    {
-      method: "POST",
-      body: JSON.stringify(payload)
-    }
-  );
-  return res.json();
-};
-
-const fetchMobilePayload = () =>
-  getInAppPayload({
-    events: [
-      {
-        query: {
-          personalization: {
-            surfaces: [surface]
-          }
-        },
-        xdm: {
-          timestamp: new Date().toISOString(),
-          implementationDetails: {
-            name: "https://ns.adobe.com/experience/mobilesdk/ios",
-            version: "3.7.4+1.5.0",
-            environment: "app"
-          }
-        }
-      }
-    ]
-  });
-
 export default function InAppMessages() {
   const [sentEvent, setSentEvent] = useState(false);
   const [customTraitKey, setCustomTraitKey] = useState("");
   const [customTraitValue, setCustomTraitValue] = useState("");
 
+  let unsubscribeRulesetItems = () => undefined;
+
+  useEffect(() => {
+    alloyInstance("subscribeRulesetItems", {
+      callback: result => {
+        console.log("subscribeRulesetItems", result);
+      }
+    }).then(({ unsubscribe }) => {
+      unsubscribeRulesetItems = () => unsubscribe();
+    });
+
+    return () => {
+      unsubscribeRulesetItems();
+    };
+  }, []);
+
```

Comment:
There is a race condition here if React tries to unsubscribe before "subscribeRulesetItems" completes. This might be a better pattern for documentation:

```
useEffect(() => {
  const unsubscribePromise = alloyInstance("subscribeRulesetItems", {
    callback: result => {
      console.log("subscribeRulessetItems", result);
    }
  });
  return () => {
    unsubscribePromise.then(unsubscribe => unsubscribe());
  }
,[]);
```

Reply:
good catch